qt/python/mantidqt/widgets/sliceviewer/view.py

```diff
@@ -67,6 +68,23 @@ def plot(self, ws, **kwargs):
         self.mpl_toolbar.update() # clear nav stack
         self.canvas.draw_idle()
 
+    def plot_matrix(self, ws, **kwargs):
+        """
+        clears the plot and creates a new one using a MatrixWorkspace
+        """
+        self.ax.clear()
+        if use_imshow(ws):
+            self.im = self.ax.imshow(ws, origin='lower', aspect='auto',
+                                     transpose=self.dimensions.transpose,
+                                     norm=self.colorbar.get_norm(), **kwargs)
+        else:
+            self.im = self.ax.pcolormesh(ws, transpose=self.dimensions.transpose,
+                                         norm=self.colorbar.get_norm(), **kwargs)
+        self.ax.set_title('')
+        self.colorbar.set_mappable(self.im)
+        self.mpl_toolbar.update() # clear nav stack
+        self.canvas.draw_idle()
+
```

Comment:
May we avoid this level of code duplication? Maybe adding a check into `use_imshow` or passing `WS_TYPE`?